src/components/resizable.tsx

```diff
@@ -0,0 +1,164 @@
+    'resizable-width-change' event.
+
+    For usage refer to the recordset.tsx component
+*/
+
+
+const vwToPx = (value: number) => {
+    const e = document.documentElement;
+    const g = document.getElementsByTagName('body')[0];
+    const x = window.innerWidth || e.clientWidth || g.clientWidth;
+
+    const result = (x * value) / 100;
+    return result;
+}
+
+type LeftPaneProps = {
+    children: (ref: React.RefObject<HTMLDivElement>) => JSX.Element,
+    leftWidth: number | undefined,
+    setLeftWidth: (value: number) => void
+};
+
+const LeftPane = ({ children, leftWidth, setLeftWidth }: LeftPaneProps): JSX.Element => {
+    const leftRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (leftRef?.current) {
+            if (!leftWidth) {
+                setLeftWidth(leftRef.current.clientWidth);
+                return;
+            }
+
+            fireCustomEvent('resizable-width-change', '.split-view', { width: leftWidth });
+            leftRef.current.style.width = `${leftWidth}px`;
+        }
+    }, [leftRef, leftWidth, setLeftWidth]);
+
+    return children(leftRef);
+};
+
+
+type SplitViewProps = {
+    left: (ref: React.RefObject<HTMLDivElement>) => JSX.Element,
+    right: React.ReactNode,
+    className?: string,
+    minWidth?: number,
+    maxWidth?: number,
+    initialWidth?: number,
+    convertMinWidth?: boolean,
+    convertMaxWidth?: boolean,
+    convertInitialWidth?: boolean
+};
+
+const SplitView = ({
+    left,
+    right,
+    className,
+    minWidth = 250,
+    maxWidth = 450,
+    initialWidth = 270,
+    convertMinWidth = false,
+    convertMaxWidth = false,
+    convertInitialWidth = false
+}: SplitViewProps): JSX.Element => {
+
+
+    let convertedMinWidth = minWidth;
+    let convertedMaxWidth = maxWidth;
+    let convertedInitialWidth = initialWidth;
+
+    if (convertMinWidth) {
+        convertedMinWidth = vwToPx(minWidth);
+    }
+
+    if (convertMaxWidth) {
+        convertedMaxWidth = vwToPx(maxWidth);
+    }
+
+    if (convertInitialWidth) {
+        convertedInitialWidth = vwToPx(initialWidth);
+    }
+
+    const [leftWidth, setLeftWidth] = useState<undefined | number>(convertedInitialWidth || undefined);
+    const [separatorXPosition, setSeparatorXPosition] = useState<undefined | number>(undefined);
+    const [dragging, setDragging] = useState(false);
+
+    const onMouseDown = (e: React.MouseEvent) => {
+        setSeparatorXPosition(e.clientX);
+        setDragging(true);
+    };
+
+    const onMouseMove = (e: MouseEvent) => {
+        e.preventDefault();
+        onMove(e.clientX);
+    };
+
+    const onMouseUp = () => {
+        setDragging(false);
+    };
+
+    const onMove = (clientX: number) => {
+        if (dragging && leftWidth && separatorXPosition) {
+            const newLeftWidth = leftWidth + clientX - separatorXPosition;
+            setSeparatorXPosition(clientX);
```

Comment:
I'm noticing some jittering while resizing the panel. Could the reason be the fact that you're setting two state variables and you're not in control of their batching?

In general, I think it might be better to merge these two state variables into an object.

Reply:
fixed. Bunching all these states together into a single state did the trick.